pkg/bootkube/status.go

```diff
@@ -0,0 +1,123 @@
+package bootkube
+
+import (
+	"fmt"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_2"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	appKey       = "k8s-app"
+	doesNotExist = "DoesNotExist"
+)
+
+func CheckStatus(pods []string, timeout time.Duration) error {
```

Comment:
nit: Just a naming nit, but feels like `WaitUntilPodsRunning` would be more descriptive than `CheckStatus`, since this function doesn't return statuses and just blocks until all pods are running or timeout is reached.